### DISCUSSION TITLE
Install the custom program on the hub.

### DISCUSSION THREAD

**Discussion by sparmar18** [pybricks/support#1965] [Q&A]
Hi, I’m new to the Lego world and don’t have much knowledge about how the hub program works. I have a Lego Technic Hub and a custom program for our company's model of a telescopic conveyor. The person who designed the program is no longer with us, and I need help with the steps to install that program on the Hub using Pybricks. Can someone please provide me with the steps?

**Comment by laurensvalk** [comment]
You can follow the starter guide here: https://pybricks.com/learn/

It sounds like you can jump to section 1.2 on the left hand table of contents since you already have all your materials.

**Reply by laurensvalk** [reply]
If you have a video of the end result, we'd love to see it :smile:

**Reply by sparmar18** [reply]
> You can follow the starter guide here: https://pybricks.com/learn/
> 
> It sounds like you can jump to section 1.2 on the left hand table of contents since you already have all your materials.

Hi, I have a program in a .py file format. Can I upload this directly to the hub? Sorry, I couldn’t find the steps to upload this type of file.

**Reply by sparmar18** [reply]
> If you have a video of the end result, we'd love to see it 😄

I will surely share that here. :)

**Reply by laurensvalk** [reply]
> > You can follow the starter guide here: https://pybricks.com/learn/
> > It sounds like you can jump to section 1.2 on the left hand table of contents since you already have all your materials.
> 
> Hi, I have a program in a .py file format. Can I upload this directly to the hub? Sorry, I couldn’t find the steps to upload this type of file.

Scrolling slightly further down the page, you'll find how to manage files. It's not all that different from most code editors.

![image](https://github.com/user-attachments/assets/0d81e323-7da8-4e91-989e-44f115471e69)

Once you've opened the script, connect to the robot and run it. (And install the firmware if your colleague hadn't done so already).

**Comment by sparmar18** [comment]
https://github.com/user-attachments/assets/04da28fa-6151-4dc5-9500-7d207bf11b92

Thanks for the help :). Here is the video of our model.

**Reply by laurensvalk** [reply]
Glad it worked. Thanks for sharing. Nice to see Pybricks being used like this!
